### ISSUE TITLE
SDK Review - Release Planner App 

### ISSUE THREAD

**Issue by ladonnaq** [Azure/azure-sdk-tools#4599]
**Approach for current release:** 
For APEX scope we will know if it is first initial beta client library because we have determined during onboarding and the release plan is created for new initial beta. If it is a new service or a service with no existing client libraries (SDKs) - then it will be initial Beta client libraries. Also, during onboarding we will confirm data plane scope and will determine if it is a new client library or an update to an existing client library. I have a field in the ADO item that I am using to track this info now - Data plane SDK release type is the field. This could be different for non-APEX or if the architecture board decides that they want a team to release a new initial beta vs a new beta or stable SDK version. But I think we can go with what I recommended in this GitHub issue for now - https://github.com/Azure/azure-sdk-tools/issues/4598. We will circle back around to this in the re-design of the UI for the SDK release app. I am going to clean up this GitHub issue by referencing in the description that for now we will use GitHub issue https://github.com/Azure/azure-sdk-tools/issues/4598 for current UI and then we will re-visit when re-designing the UI for the SDK release app.

**Comment by ladonnaq** [comment]
@kyle-patterson  Please review and edit accordingly.  I believe this is what we discussed for Beta V2 implementation for SDK Reviews.  We can have the form accessible outside of a release plan so that you can use this approach for other SDK Reviews.  This is what we have done with the Onboard new product questionnaire.  There should not be a need for it when we support all release plans but for now it needs to be a separate standalone task (outside of the release plan).  Alternatively, we could have the questionnaire as part of the scheduling app and a requirement before scheduling synchronous SDK Reviews. 



![Image](https://user-images.githubusercontent.com/67656762/199543707-0f7e78ba-352f-4105-9a62-32e5bab5de4a.png)

**Comment by ladonnaq** [comment]
@kyle-patterson @ronniegeraghty I need your input so we can provide requirements to engineering.  It is possible to initiate the SDK review by a form which will create a GitHub issue with tags/labels you require.  The guidance can explain that API reviews are electronic by default and if a in-person SDK review is required that the helpdesk will reach out to provide information for scheduling the in-person SDK Review (you can direct them to the scheduling tool).  Also, provide information on what languages require in-person SDK review, pre-requisites they need to complete, etc.

**Comment by live1206** [comment]
Info from LaDonna:

There are two meeting types for Arch Board - Introduction and SDK Review. New services and new client libraries require this meeting type. Existing client libraries (new version) can schedule the SDK review (no intro meeting required).

We will need new logic added to scheduling tool. We will know which meeting type to schedule, so we can let the scheduling tool know and the meeting type can be set. This way it is automatic, and the user doesn't have to figure out which one to choose.

**Comment by maririos** [comment]
> We will need new logic added to scheduling tool. 

The Scheduling tool already supports the 2 types of meetings. The user can select the type when scheduling. See:

<img width="396" alt="image" src="https://user-images.githubusercontent.com/9868623/217967348-eeb76fdc-0baa-4e17-9177-772a1a268830.png">

I agree with you that because we know where they are in the process and what type of meeting they need, the Release Planner should send them to the Scheduler with the type of meeting already selected. This can be accomplished by passing a value to the `Event.SubType` column. Values are `Introduction` or `Follow up`

**Comment by zzvswxy** [comment]
> Values are `Introduction` or `Follow up`

So As we discuss before, `Introduction`  type will be used in API Readiness app, and SDK release app will use `Follow up`? 
@live1206  is it correct？

**Comment by live1206** [comment]
> > Values are `Introduction` or `Follow up`
> 
> So As we discuss before, `Introduction` type will be used in API Readiness app, and SDK release app will use `Follow up`? @live1206 is it correct？

I am not quite sure about the process here, as I heard the first beta sdk will need sdk introduction meeting, version afterwards will need sdk review meeting, which is follow-up. 
Regarding where to add these meetings, @josefree could you help clarify this? Thanks!

**Comment by josefree** [comment]
About "Architecture Board Review Request - API Review/API Sign Off", I suggest to place it here along with the API Readiness milestone
![image](https://user-images.githubusercontent.com/18413981/218942411-08d1bfff-1903-48ea-bdef-9ad2985e4e76.png)
The [Architecture Board review](https://github.com/Azure/azure-sdk/issues/new?assignees=kyle-patterson%2C+ronniegeraghty&labels=architecture%2C+board-review&template=adp_api_review_and_signoff.md&title=Board+Review%3A+%3Cclient+library+name%3E) is to ensure your service libraries produced solely by a code generator follows the [Track 2 library design guidelines and specific language guidelines](https://azure.github.io/azure-sdk/general_introduction.html). 

@ladonnaq what do you think?  BTW, I wonder after service team create the issue, do they still need to use scheduling app to schedule a meeting? If not, it will be only optional, not schedule Architect Board Review in the step should not block 'Next'

**Comment by ladonnaq** [comment]
> 

1. Let's move the Arch Board after the after the Define data plane release date.  We should let them know that it will take 4-5 weeks to get on the calendar so they can schedule now.  They should have generated SDKs and passed API Readiness.  
2. If new initial Beta release, then Information Meeting with the Arch Board.  New Service or New client library. 
3. If new version of Beta release, then SDK review.

**Comment by live1206** [comment]
@ladonnaq We would like to remain the order of SDK Arch Board Review as it is because of below reasons:
- We can only detect first beta information based on the generated version from SDK grow-up step.
- Heard from @lirenhe that, Arch board needs more information during the review meeting, such as hero scenario test, which is also generated during the SDK grow-up step.

**Comment by zzvswxy** [comment]
> * We can only detect first beta information based on the generated version from SDK grow-up step.

![image](https://user-images.githubusercontent.com/20081898/219302624-470b037e-ca41-4497-b771-fe999cb87965.png)
If one of the SDK are first beta release, will set the meeting sub type to "Introduction". In other cases, will set the meeing sub type to "Follow up".

CC @ladonnaq

**Comment by ladonnaq** [comment]
> > * We can only detect first beta information based on the generated version from SDK grow-up step.
> 
> ![image](https://user-images.githubusercontent.com/20081898/219302624-470b037e-ca41-4497-b771-fe999cb87965.png) If one of the SDK are first beta release, will set the meeting sub type to "Introduction" automatically. In other cases, will set the meeing sub type to "Follow up".
> 
> CC @ladonnaq

@zzvswxy @live1206 
For APEX scope we will know if it is first initial beta client library because we have determined during onboarding and the release plan is created for new initial beta. If it is a new service or a service with no existing client libraries (SDKs) - then it will be initial Beta client libraries.  Also, during onboarding we will confirm data plane scope and will determine if it is a new client library or an update to an existing client library.  I have a field in the ADO item that I am using to track this info now - Data plane SDK release type is the field. This could be different for non-APEX or if the architecture board decides that they want a team to release a new initial beta vs a new beta or stable SDK version. But I think we can go with what I recommended in this GitHub issue for now - https://github.com/Azure/azure-sdk-tools/issues/4598.  We will circle back around to this in the re-design of the UI for the SDK release app.  I am going to clean up this GitHub issue by referencing in the description that for now we will use GitHub issue #4598 for current UI and then we will re-visit when re-designing the UI for the SDK release app.
